packages/state-transition/src/cache/epochCache.ts

```diff
@@ -748,9 +771,74 @@ export class EpochCache {
     return isAggregatorFromCommitteeLength(committee.length, slotSignature);
   }
 
+  /**
+   * Return finalized pubkey given the validator index.
+   * Only finalized pubkey as we do not store unfinalized pubkey because no where in the spec has a
+   * need to make such enquiry
+   *
+   */
+  getPubkey(index: ValidatorIndex): PublicKey {
+    return this.index2pubkey[index];
```

Comment:
This is type unsafe, may return undefined. Add a condition to check index is within bounds

Reply:
> This is type unsafe, may return undefined. Add a condition to check index is within bounds

I think the current callers use `index2pubkey[I]` directly, so I don’t think we need to explicitly check if index is within bounds. Will change the return type to `PublicKey | undefined`